sycl/include/sycl/ext/oneapi/experimental/address_cast.hpp

```diff
@@ -60,9 +56,8 @@ multi_ptr<ElementType, Space, access::decorated::no>
 dynamic_address_cast(ElementType *Ptr) {
   using ret_ty = multi_ptr<ElementType, Space, access::decorated::no>;
 #ifdef __SYCL_DEVICE_ONLY__
-  // TODO: Remove this restriction.
```

Comment:
This comment is removed, but I don't really see any restriction being removed. Can you explain?

Reply:
See comments starting with https://github.com/intel/llvm/pull/15423#discussion_r1765696303, plus the description of this PR.